### ISSUE TITLE
esp32: When creating an UART object an error message pops up.

### ISSUE THREAD

**Issue by robert-hh** [micropython/micropython#11872]
This is the sequence of commands at REPL. The board is a "classic" ESP32 with or without SPIRAM.  UART works. No unexpected message with a C3, S2 and S3 board.

```
MicroPython v1.20.0-243-gd93316fe3 on 2023-06-23; ESP32 module (spiram) with ESP32
Type "help()" for more information.
>>> from machine import UART, Pin
>>> u=UART(1, rx=4, tx=5)
E (17470) uart: uart_param_config(726): Invalid src_clk

>>> import platform
>>> platform.platform()
'MicroPython-1.20.0-xtensa-IDFv5.0.2-with-newlib4.1.0'
>>> 
```

**Comment by dpgeorge** [comment]
Thanks for the report, I can confirm the issue.  I have a fix, to always set `source_clk` in the uart config struct.

**Comment by dpgeorge** [comment]
Fixed by 5a9a0df3407cb195ff1f61cf9cba0fc8700c54c6
